src/qt/bitcoinamountlabel.cpp

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) 2019 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <qt/bitcoinamountlabel.h>
+
+BitcoinAmountLabel::BitcoinAmountLabel(QWidget* parent)
+    : QLabel(parent)
+{
+}
+
+void BitcoinAmountLabel::setText(const QString& newText)
+{
+    if (cache.isEmpty()) {
+        QLabel::setText(newText);
+    } else {
+        cache = newText;
+    }
+}
+
+void BitcoinAmountLabel::changePrivacyMode(bool privacy)
+{
+    if (privacy && cache.isEmpty()) {
+        cache = text();
+        QLabel::setText("h-i-d-d-e-n");
```

Comment:
I think a fence of `#` translates better.

Couldn't this use `BitcoinUnits::format(unit, 0).replace("0", "#") + unit.shortName()` (or whatever is used right now), so that the UI doesn't reflow the columns on every toggle?

Reply:
![Screenshot from 2019-07-22 08-04-35](https://user-images.githubusercontent.com/6399679/61631080-66e61e00-ac57-11e9-9143-34c0965c3f89.png)

Reply:
> I think a fence of # translates better.

Done.

> ... so that the UI doesn't reflow the columns on every toggle?

This requires a monospaced font.

Reply:
> This requires a monospaced font.

Done.